Adding ScalarType and Layout type tag in IValue

charliechen0401 opened the pull request:
Summary:
It's basically step 2 in https://github.com/pytorch/pytorch/pull/30694#issuecomment-564205276.

Right now, just forcing those types to return intType or following behavior of IntType now.

Test Plan: Waiting for all tests to pass

Differential Revision: D18914912

kostmo commented:
## CircleCI build failures summary
As of commit 7e81970:
* **1/1** broken upstream at merge base 643ca5d (see [grid view](https://dr.pytorch.org/master-timeline.html?count=1&sha1=643ca5def2ea140f16bb064740b8de331ee465d8&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35))
    * You may want to rebase on the `viable/strict` branch (see its [recency history](https://dr.pytorch.org/master-viable-commits.html)):
        * If your commit is newer than `viable/strict`, you can try basing on an older, stable commit:
          ```
          git fetch viable/strict
          git rebase --onto viable/strict $(git merge-base origin/master HEAD)
          ```
        * If your commit is older than `viable/strict`:
          ```
          git fetch viable/strict
          git rebase viable/strict
          ```
* **0/1** failures introduced in this PR



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=7e8197022494734cd174afdc8a1b1533a545eed1).

### 1 upstream failure recognized by patterns:
These builds matched patterns, but were probably caused by upstream breakages:
* [caffe2_onnx_py2_gcc5_ubuntu16_04_test](https://dr.pytorch.org/build-details.html?build_id=75484928)



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

Commit:
Adding ScalarType and Layout type tag in IValue (#31338)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/31338

It's basically step 2 in https://github.com/pytorch/pytorch/pull/30694#issuecomment-564205276.

Right now, just forcing those types to return intType or following behavior of IntType now.

Test Plan: Waiting for all tests to pass

Differential Revision: D18914912

fbshipit-source-id: 0086d16e331d6401bbb835d1e2c8194fd740f565
